File: Intersection-of-Arrays-with-Distinct.js
/**
 * @param {number[]} a
 * @param {number[]} b
 * @returns {number}
 */
class Solution {
    intersectSize(a, b) {
        // code here
        a.sort(function(x, y) { return x - y; });
        b.sort(function(x, y) { return x - y; });
        let iA=[]
        let i=0
        let j=0
        
        while (i<a.length && j<b.length){
            if (a[i]<b[j]){
                i++
            }
            else if (a[i]>b[j]){
                j++
            }
            else{
                iA.push(a[i])
                i++
                j++
            }
        }
        return iA.length
    }
}

File: Intersection-of-Arrays-with-Distinct.py
class Solution:
    def intersectSize(self,a, b):
        # code here
        (a.sort())
        (b.sort())
        iA=[]
        i,j=0,0
        while  i<len(a) and j<len(b):
            if a[i]<b[j]:
                i+=1 
            elif a[i]>b[j]:
                j+=1 
            else:
                iA.append(a[i])
                i+=1
                j+=1
        
        return len(iA)
